api.ts

```diff
@@ -1423,23 +1440,46 @@ export function makeMetadataFormula(
   });
 }
 
-function makePropertyAutocompleteFormula(
-  execute: PropertyAutocompleteMetadataFunction,
-  options?: {connectionRequirement?: ConnectionRequirement},
-): PropertyAutocompleteMetadataFormula {
+/**
+ * Builds a formula to store in {@link SyncTableAutocompleters}.
+ *
+ * @hidden
+ */
+export function makePropertyAutocompleteFormula<SchemaT extends Schema>({
+  execute,
+  schema,
+  name,
+}: {
+  execute: PropertyAutocompleteMetadataFunction<Array<SchemaType<SchemaT>>>;
+  schema: SchemaT;
+  name: string;
+}): PropertyAutocompleteMetadataFormula<SchemaT> {
   if (!(execute instanceof Function)) {
-    throw new Error(`Value for propertyAutocomplete must be a function`);
+    throw new Error(`Value for execute must be a function`);
   }
-  return makeObjectFormula({
-    name: 'getPropertyAutocompleteMetadata',
-    description: 'Gets property autocomplete',
-    execute([], context) {
-      return execute(context as PropertyAutocompleteExecutionContext);
-    },
+
+  type ResultT = SchemaType<ArraySchema<SchemaT>>;
+
+  // The type SchemaType<ArraySchema<T>> is equivalent to Array<SchemaType<T>>, but typescript doesn't know
+  // that unless we do a cast.
```

Comment:
:(

Reply:
Earlier iterations of this SDK code with different API surfaces had much sadder typescript casts, so I was actually kind of happy that this was the only weird one

However... when we add the ability to run a `cacheTtl` in addition to the result array, the types here will be an imperfect reflection of what the formula can actually return.

If we want to allow both a simple scenario where people just return a raw array as well as a more complex scenario where you can return that array wrapped in an object with additional information like a `cacheTtl`, that'll be too sophisticated for the pack schema types to express natively.